aws/sdk/integration-tests/s3/tests/mandatory-query-param.rs

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+use aws_smithy_client::never::NeverConnector;
+
+#[tokio::test]
+#[should_panic(
+    expected = r#"ConstructionFailure(MissingField { field: "upload_id", details: "cannot be empty or unset" })"#
+)]
+async fn test_mandatory_query_param_is_unset() {
+    let conf = aws_sdk_s3::Config::builder().build();
+    let conn = NeverConnector::new();
+
+    let client = aws_sdk_s3::Client::from_conf_conn(conf, conn.clone());
+
+    client
+        .abort_multipart_upload()
+        .bucket("a-bucket")
+        .key("a-key")
+        .send()
+        .await
+        .unwrap();
+}
```

Comment:
I updated these tests to work differently. After talking to Chase, I implemented it so that, for required query params, the key will always be written, regardless of if a value was set. This allows S3 to differentiate the request correctly and reject it for a missing/empty upload ID.

Reply:
The updated tests now check that the uploadId key is in the sent request